modules/stats.php

```diff
@@ -370,7 +370,16 @@ function stats_reports_page() {
 	if ( isset( $_GET['dashboard'] ) )
 		return stats_dashboard_widget_content();
 
-	$blog_id = stats_get_option( 'blog_id' );
+	if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+		$blog_id = get_current_blog_id();
+		$bloginfo = get_blog_details( (int) $blog_id );
+		$domain = $bloginfo->domain;
+	} else {
+		$blog_id = Jetpack_Options::get_option( 'id' );
+		$url = home_url();
+		$url_parts = parse_url( $url );
+		$domain = $url_parts['host'];
```

Comment:
We have a handly method `Jetpack::build_raw_urls( $url )` that we can use here.  

`Jetpack::build_raw_urls( get_home_url() )` should do the trick.